src/spawnable/projectile/behavior.rs

```diff
@@ -201,9 +203,21 @@ fn deal_damage_on_contact(
                 {
                     // deal damage to mob
                     audio_channel.play(audio_assets.bullet_ding.clone());
-                    for (mob_entity_q, mut mob_component) in mob_query.iter_mut() {
-                        if *mob_entity == mob_entity_q {
+                    for (mob_entity_q, mob_transform, mut mob_component) in mob_query.iter_mut() {
+                        if *mob_entity == mob_entity_q && *projectile_damage > 0.0 {
                             mob_component.health.take_damage(*projectile_damage);
+
+                            // spawn damage dealt text effect
+                            spawn_effect_event_writer.send(SpawnEffectEvent {
+                                effect_type: EffectType::Text(TextEffectType::DamageDealt),
```

Comment:
Let's look at why there are 4 of these. Maybe some polymorphism would help here. Maybe under misc/health.rs?